### PR TITLE
Migration Guide for the documentation

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/migration.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
...urce, the number of partitions in the returned RDD is determined  by the DataSource. In the Spark 2.4 version of the Cosmos DB connector the DataSource would create 1 Spark partition for each physical Cosmos DB partition. Especially for smaller Cosmos accounts this often lead to situations where the number of resulting Spark partitions was lower than the number of Cores available for the executors - so the latency of the spark job was relatively high because not all Executors would participate in processing teh read operations from Cosmos. The main reason for this decision was that it wasn't possible to scope a query to just a fragment of a physical partition when the Cosmos DB Connector for Spark 2.4 was created. With the new connector for Spark 3 we now by default use a new capability available in the Cosmos DB Backend, and the Java V4 SDK that allows scoping queries to just a fragment of a physical partition. As a result the default partitioning strategy will ensure that at least as many Spark partitions as Executor Cores are created. The number of Spark partition created will depend on the number of available Cores for executors, and the storage size in each physical partition.
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/migration.md:54]: typo: teh -> the

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/migration.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+### Structured streaming
+The API for Structured Streaming between DataSource V1 and V2 has changed. The new Cosmos DB connector uses teh DataSource V2 API. As a result, some proprietary approaches (like persisting the progress/bookmarks/offset in a Cosmos DB container) are replaced by default Spark mechanics where possible. So the new connector for example just uses Spark Offsets to expose the progress. Spark will store these offsets in the provided checkpoint location and when you want to restart/recover a query the mechanism is teh same as with any other DataSource supporting Structured Streaming - like Kafka for example. (The Structured Streaming Programming Guide)[http://spark.apache.org/docs/latest/structured-streaming-programming-guide.html] is a good starting point to learn how to work with Structured Streaming.
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/migration.md:61]: typo: teh -> the

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/migration.md (hunk 1 above)
Comment (ealsur) [inline sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/migration.md:54]: nit: Extra spaces `etermined  by the DataSo`

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/migration.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+### "One DataSource rules them all" vs. separate DataSource for ChangeFeed
+In the Cosmos DB Connector for SPark 2.4 all operations (writing or reading documents/items from the container as well as processing change feed) were surfaced by one DataSource. With the new COsmos DB Connector for Spark 3 we are using two different DataSources - `cosmos.oltp.items` and `cosmos.oltp.changefeed` to be able to express the supported capabilities in the DataSource V2 API correctly - for example to expose to the Spark runtime that the change feed DataSource would not support writes but supports read stream operations - while the DataSource for items support writing (both streaming and batch) but no read stream operations. This means when migrating existing notebooks/programs, it will be necessary to not only change the configuration but also the identifier in the `format(xxx)` clause to use the right identifier - `cosmos.oltp.items` for operations on items and `cosmos.oltp.changefeed` when processing change feed events.
```
Comment (ealsur) [inline sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/migration.md:58]: nit: `SPark 2.4 `

Comment (ealsur) [reply sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/migration.md:58]: nit: `COsmos DB Connector`

**Review by ealsur — APPROVED** [review]

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/migration.md (hunk 1 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/migration.md:54]: Thanks - fixed

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/migration.md (hunk 2 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/migration.md:61]: Fixed

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/migration.md (hunk 1 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/migration.md:54]: Fixed

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/migration.md (hunk 3 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/migration.md:58]: Fixed

**Pull request merged**